### DISCUSSION TITLE
Question about the license and the schema files

### DISCUSSION THREAD

**Discussion by pjohnmeyer** [Open-Cap-Table-Coalition/Open-Cap-Format-OCF#111] [Q&A] [answered]
The license currently reads:

> Permission to copy and distribute OCF Files, in any medium for any purpose and without fee or royalty is hereby granted, provided that you include the following on ALL copies of an OCF File, or portions thereof, that you use:
> * A link to the original OCF File.
> * A notice of the form: "Copyright © [$date-of-ocf-file] [Open Cap Table Coalition](https://opencaptablecoalition.com/)."

However, if I interpret this correctly, this is impossible to comply with if I am copying, for example, the `.json` schema files, because I can't just add a comment to the top of a `.json` file -- because that's invalid JSON. Am I interpreting this correctly?

**Comment by JSv4** [comment]
Good point. Possibly we just add a field to our base object?:

`"Copyright": "Copyright © [$date-of-ocf-file] Open Cap Table Coalition @ https://opencaptablecoalition.com."`

**Reply by pjohnmeyer** [reply]
I don't know if json-schema allows arbitrary fields like that. They do have a reserved field `$comment` we could put it in, but based on the wording it would need to be in every file.

**Reply by JSv4** [reply]
Yeah, that's a better approach. The intent of the license is to have the notice in the schema files, but not implementing ocf files. I could see how the defined term "OCF File" might be confusing, but it should only apply to the schemas and markdown docs:

`By using and/or copying Open Cap Table Coalition JSON Schema files or documentation thereof (each, individually, an “OCF File”, and, collectively, the “OCF Files”)`

Sounds like the `$Comment` field would work for this purpose. Maybe we should change the defined term in the license to refer to the schemas and documentation files as a "Schema File" or, collectively, "Schema Files"?

**Reply by pjohnmeyer** [reply]
Understood; by "every file" I meant "every schema file" -- not just the base object. I did not mean the actual implementing files.

**Reply by JSv4** [reply]
Yup, that's the intent of the license. A lot of projects do that, though it's not strictly necessary. Would you prefer to leave out the comment?

**Comment by JSv4** [comment, marked as answer]
Let's add the copyright notices as a `$Comment` field in all schema files.

**Reply by pjohnmeyer** [reply]
Will do. I don't _like_ it (and it's one of those things we have to write a script to update on release which is lame), but by the terms of the license I _think_ it's needed.
